hack/generate-kcp-api.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+
+set -e
+
+if ! which kubectl-kcp; then
+  echo "kubectl-kcp required on path"
+  echo "you can install it with running:"
+  echo "    $ git clone https://github.com/kcp-dev/kcp && cd kcp && make install"
+  exit 1
+fi
+
+THIS_DIR="$(dirname "$(realpath "$0")")"
+CRD_DIR="$( realpath ${THIS_DIR}/../config/crd/bases)"
+KCP_API_DIR="$( realpath ${THIS_DIR}/../config/kcp)"
+
+KCP_API_SCHEMA_FILE_CURRENT="${KCP_API_DIR}/apiresourceschema_release.yaml"
+KCP_API_SCHEMA_FILE_NEW="${KCP_API_DIR}/apiresourceschema_release.yaml_new"
+cat << EOF > ${KCP_API_SCHEMA_FILE_NEW}
+# This file is generated from CRDs by ./hack/generate-kcp-api.sh script.
+# Please do not modify!
+
+EOF
+
+# APIResourceSchema is immutable so when we want to update something, we actually have to create new version.
+# Version is defined by this prefix, which is taken from date. This will allow us to do new version each minute, which
+# should be hopefully enough granularity :)
+PREFIX=$( TZ="Etc/UTC" date +%Y%m%d%H%M )
+
+I=0
+for CRD in $( ls ${CRD_DIR} ); do
+  kubectl-kcp crd snapshot -f "${CRD_DIR}/${CRD}" --prefix v${PREFIX} >> ${KCP_API_SCHEMA_FILE_NEW}
+done
+
+# If there are some changes in new generated file, we replace old one. Otherwise just remove new file.
+# The regex is there to ignore name change, because we're updating date there so it is expected to change.
+# Ignored line looks like this:
+# '  name: v202206151654.applications.appstudio.redhat.com'
+if ! diff -I '^  name: v[0-9]\{12\}\..*\.appstudio\.redhat\.com$' ${KCP_API_SCHEMA_FILE_CURRENT} ${KCP_API_SCHEMA_FILE_NEW} > /dev/null; then
+  mv ${KCP_API_SCHEMA_FILE_NEW} ${KCP_API_SCHEMA_FILE_CURRENT}
+  echo "updated KCP APIResourceSchema for release saved at '${KCP_API_SCHEMA_FILE_CURRENT}'"
+else
+  echo "no changes in KCP API"
+  rm ${KCP_API_SCHEMA_FILE_NEW}
```

Comment:
This should be a trap to be sure it's deleted even if the script breaks for some reason. For example, if you run this script without logging in your cluster, the file will be created with the heading, the script will break in line 31 and the file won't be deleted.